include/triton/Tools/LinearLayout.h

```diff
@@ -616,6 +625,11 @@ class LinearLayout {
   // the same, and are the bases the identity?
   bool squareSublayoutIsIdentity(ArrayRef<StringAttr> dimNames) const;
 
+  // Is the sublayout defined from dimNames to dimNames a subpermutation matrix?
+  // I.e. the layout matrix is formed by selecting unique rows from the identity
+  // matrix and adding zero rows.
+  bool squareSublayoutIsPermutation(ArrayRef<StringAttr> dimNames) const;
+
```

Comment:
Can we please create a file with helper functions? Either in this PR or a different one. We are just dumping every function as a method with looong names, and it's starting to look a bit too much like Java :P

Reply:
Yeah let me do that now. I can factor a bunch of stuff into there. Where should I put this file?

Reply:
(althought if the diff is small that might be alright)